### PR TITLE
Rename storage.aggregation_workers_number to parallel_operations

### PR DESCRIPTION
**Pull request by jd**
The option storage.aggregation_workers_number is not only used by storage, but
also by API cross aggregation. Make it generic and move the map_in_thread
method in a generic parallel_map function in gnocchi.utils.

### CODE REVIEW

**Comment by jd** [comment]
Pull-request updated, HEAD is now 6b5636ec2a5e84818ebe65f28b1796ab968343ec

**Comment by jd** [comment]
Pull-request updated, HEAD is now 899c258196b9ecd339ff5141a48062b9ce9503dc

File: gnocchi/tests/test_utils.py [hunk 1]
```diff
@@ -115,3 +115,17 @@ def test_context_manager(self):
...
+
+class ParallelMap(tests_base.TestCase):
+    def test_parallel_map_one(self):
+        utils.parallel_map.NUM_WORKERS = 1
+        self.assertEqual([1, 2, 3],
+                         utils.parallel_map(lambda x: x,
+                                            [[1], [2], [3]]))
+
+    def test_parallel_map_four(self):
+        utils.parallel_map.NUM_WORKERS = 4
```
Comment (chungg) [inline gnocchi/tests/test_utils.py]: these are some "good" tests

**Comment by jd** [comment]
Just rebased

**Comment by jd** [comment]
Rebased, conflict on processor.py

**Pull request merged**